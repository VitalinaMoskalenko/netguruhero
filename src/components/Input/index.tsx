import React, { ChangeEvent } from "react";
import styled, { CSSProperties } from "styled-components";
import { SmallBody } from "../Paragraphs";

type PropsType = {
  label: string;
  value?: string;
  onChange: (value: string) => void;
  containerStyle?: CSSProperties;
  errorMessage?: string;
};

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

type FieldPropsType = {
  isError: boolean;
};

const Field = styled.input<FieldPropsType>`
  flex: 1;
  background-color: ${({ theme }) => theme.colors.white};
  border-radius: 8px;
  height: 40px;
  border: ${({ isError, theme }) =>
    isError ? `1px solid ${theme.colors.red}` : "0px"};
  padding: 16px;
  font-family: "Courier New", Courier, monospace;
`;

const Label = styled(SmallBody)`
  font-weight: bold;
  margin-bottom: 6px;
  opacity: 0.5;
`;

const ErrorMessage = styled(SmallBody)`
  margin-top: 4px;
  color: ${({ theme }) => theme.colors.red};
`;

const Input = ({
  label,
  value,
  onChange,
  containerStyle,
  errorMessage,
}: PropsType) => {
  const onChangeValue = (e: ChangeEvent<HTMLInputElement>) => {
    onChange(e.target.value);
  };

  return (
    <Container style={containerStyle}>
      <Label>{label}</Label>
      <Field isError={!!errorMessage} value={value} onChange={onChangeValue} />
      {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
    </Container>
  );
};

export default Input;
