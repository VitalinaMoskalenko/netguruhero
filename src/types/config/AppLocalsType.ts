export enum AppLocalsType {
  en = "en",
}
