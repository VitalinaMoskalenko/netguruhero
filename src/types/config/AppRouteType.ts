export enum AppRouteType {
  home = "/",
  addHeroForm = "/addHeroForm",
  heroDetails = "/heroDetails",
  heroDetailsWithId = "/heroDetails/:id",
}
