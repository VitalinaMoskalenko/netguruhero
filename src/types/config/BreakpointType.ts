export enum BreakpointType {
  xxs = 375,
  xs = 768,
  xxl = 1440,
}
