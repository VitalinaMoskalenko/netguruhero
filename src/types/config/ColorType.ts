export enum ColorType {
  white = "white",
  black = "black",
  green = "#63cd93",
  blue = "#6199f3",
  gray = "#f5f6fb",
  black05 = "rgba(0,0,0, 0.5)",
}
