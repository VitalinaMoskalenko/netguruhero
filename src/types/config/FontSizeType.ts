export enum FontSizeType {
  smallBody = 14,
  body = 18,
  h5 = 16,
  h4 = 18,
  h3 = 24,
  h2 = 32,
  h1 = 48,
}
