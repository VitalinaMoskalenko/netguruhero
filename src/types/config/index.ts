export * from "./AppRouteType";
export * from "./FontSizeType";
export * from "./ColorType";
export * from "./BreakpointType";
export * from "./AppLocalsType";
