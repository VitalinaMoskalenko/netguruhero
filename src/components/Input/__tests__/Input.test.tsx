import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import Input from "../";
import { theme } from "../../../config";

it("renders input correctly", () => {
  const inputComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Input label="" value="" onChange={() => {}} />
      </ThemeProvider>
    )
    .toJSON();

  expect(inputComponent).toMatchSnapshot();
});
