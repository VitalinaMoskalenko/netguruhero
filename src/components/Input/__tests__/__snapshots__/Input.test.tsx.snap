// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input correctly 1`] = `
<div
  className="sc-dlnjwi koSwQA"
>
  <p
    className="sc-gtsrHT sc-eCApnc ckjQab jDEXhY"
  >
    
  </p>
  <input
    className="sc-hKFxyN iMFJdb"
    onChange={[Function]}
    value=""
  />
</div>
`;
